features/machine/autoscaler.feature

```diff
@@ -175,3 +175,27 @@ Feature: Cluster Autoscaler Tests
     Then the step should succeed
     And the output should match "Annotations:\s+<none>"
 
+  # @author zhsun@redhat.com
+  # @case_id OCP-23745
+  @admin
+  @destructive
+  Scenario: Machineautoscaler can be deleted when its referenced machineset does not exist
+    Given I have an IPI deployment
+    And I switch to cluster admin pseudo user
+
+    Given I store the number of machines in the :num_to_restore clipboard
+    And admin ensures node number is restored to "<%= cb.num_to_restore %>" after scenario
+
+    Given I use the "openshift-machine-api" project
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/cloud/machine-autoscaler.yml" replacing paths:
+      | ["metadata"]["name"]               | maotest |
+      | ["spec"]["minReplicas"]            | 1       |
+      | ["spec"]["maxReplicas"]            | 3       |
+      | ["spec"]["scaleTargetRef"]["name"] | invalid |
+    Then the step should succeed
+    And admin ensures "maotest" machineautoscaler is deleted after scenario
```

Comment:
I think we can remove this line. Because the falling step also deletes the MAO and verifies the deletion.

Reply:
Other than that looks all good!

Reply:
This is the actual step being tested, so I'd prefert it be deleted in this scenario(not after). Remove the `after scenario` looks better.

Reply:
So if it is testing delete we use delete command, if it is clean up environment we use `after scenario`.

Reply:
@pruan-rht Can you please advise? We want to test the deletion is successful here. Should we be using `I run the delete ..` or `admin ensures ... `?

Reply:
@liangxia Can you please advise?

Reply:
Generally ,we do prepare/configuration in Before Scenario. And clean-up in After Scenario.
`ensure xxx is deleted` actually do a `oc delete` to trigger the deletion, then do `oc get` to verify the deletion is done.
Adding `after scenario` do the same operation, the difference is that it happens in the clean-up, while without it the operations happens during the scenario execution.

For this special case, the deletion happens just after the creation, and the deletion happens to be the last step of the scenario. Both way are OK for me.

Reply:
Thanks, then it seems removing the 'after scenario' would do